File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    
    stack = []
    for item in ingredient:
        if len(stack) < 3:
            stack.append(item)
        else:
            if item == 1:
                if stack[-1] == 3 and stack[-2] == 2 and stack[-3] == 1:
                    stack.pop()
                    stack.pop()
                    stack.pop()
                    answer += 1
                else:
                    stack.append(item)
            else:
                stack.append(item)
                
    return answer

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    let answer = 0;
    let result = [];
    
    for (const item of ingredient) {
        if (result.length < 3) {
            result.push(item);
        } else {
            const stackSize = result.length;
            if (item === 1 && result[stackSize - 3] === 1 && result[stackSize - 2] === 2 && result[stackSize - 1] === 3) {
                answer += 1;
                result.pop();
                result.pop();
                result.pop();
            } else {
                result.push(item);
            }
        }
    }
    
    return answer;
}